internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -141,56 +138,24 @@ type clientgoCachedProxyResolver struct {
 	deprecatedLogger logrus.FieldLogger
 	logger           logr.Logger
 
-	// channels
-	update chan *cachedObject
-	del    chan *cachedObject
-
 	// locks
 	internalCacheLock *sync.RWMutex
 }
 
-// cacheAction indicates what caching action (update, delete) was taken for any particular runtime.Object.
-type cacheAction string
-
-var (
-	// updated indicates that this object was either newly added OR updated in the cache (no distinction made)
-	updated cacheAction = "updated"
-
-	// deleted indicates that this object was removed from the cache
-	deleted cacheAction = "deleted"
-)
-
-// cachedObject represents an object that has been processed by the cacheServer
-type cachedObject struct {
-	action cacheAction
-	err    error
-
-	key        string
-	runtimeObj runtime.Object
-}
-
 // -----------------------------------------------------------------------------
 // Client Go Cached Proxy Resolver - Public Methods - Interface Implementation
 // -----------------------------------------------------------------------------
 
 func (p *clientgoCachedProxyResolver) UpdateObject(obj client.Object) error {
-	cobj := &cachedObject{action: updated, key: p.clientObjectKey(obj), runtimeObj: obj.DeepCopyObject()}
-	select {
-	case p.update <- cobj:
-		return nil
-	default:
-		return fmt.Errorf("the proxy is too busy to accept requests at this time, try again later")
-	}
+	p.internalCacheLock.Lock()
```

Comment:
Isn't this overlap to https://github.com/Kong/kubernetes-ingress-controller/pull/1612/files#diff-b26d414d0eb20deebca0d2b785bd3e1090aa1a7a182e7dc30640d68c53cb40daR236, which introduce lock overcost ?  actionable:  keep only one of them.

Reply:
You're right! I forgot I had implemented this two different ways. I'll remove the higher level one (this one) :+1: